pkg/capabilities/consensus/ocr3/reporting_plugin.go

```diff
@@ -143,6 +146,15 @@ func (r *reportingPlugin) Outcome(outctx ocr3types.OutcomeContext, query types.Q
 			continue
 		}
 
+		for _, id := range obs.RegisteredWorkflowIds {
+			// Count how many times a workflow ID is seen from Observations
+			if _, ok := seenWorkflows[id]; ok {
```

Comment:
We need protection from malicious nodes sending a list with one ID repeated multiple times.

Reply:
What if we just make sure the workflows are not repeated on the Observation phase, or you would rather give the cleanup responsibility to Outcome instead @bolekk ?

Reply:
It has to happen here because a malicious node can send a bad observation regardless.

Reply:
I agree.